Compile error for functions with `&str` argument

```rust
/// Returns the length of the string.
#[gen_stub_pyfunction]
#[pyfunction]
fn f(x: &str) -> PyResult<usize> {
    Ok(x.len())
}
```

results in

```
error[E0277]: the trait bound `&str: pyo3::FromPyObject<'_>` is not satisfied
  --> pyo3-stub-gen-testing-pure/src/lib.rs:24:9
   |
24 | fn f(x: &str) -> PyResult<usize> {
   |         ^^^^ the trait `PyClass` is not implemented for `&str`, which is required by `&str: pyo3::FromPyObject<'_>`
   |
   = help: the following other types implement trait `pyo3::FromPyObject<'py>`:
             <(T0, T1) as pyo3::FromPyObject<'py>>
             <(T0, T1, T2) as pyo3::FromPyObject<'py>>
             <(T0, T1, T2, T3) as pyo3::FromPyObject<'py>>
             <(T0, T1, T2, T3, T4) as pyo3::FromPyObject<'py>>
             <(T0, T1, T2, T3, T4, T5) as pyo3::FromPyObject<'py>>
             <(T0, T1, T2, T3, T4, T5, T6) as pyo3::FromPyObject<'py>>
             <(T0, T1, T2, T3, T4, T5, T6, T7) as pyo3::FromPyObject<'py>>
             <(T0, T1, T2, T3, T4, T5, T6, T7, T8) as pyo3::FromPyObject<'py>>
           and 53 others
   = note: required for `&str` to implement `pyo3::FromPyObject<'_>`
```

--- a/examples/pure/pure.pyi
+++ b/examples/pure/pure.pyi
@@ -19,6 +19,12 @@ def create_dict(n:int) -> dict[int, list[int]]:
 def read_dict(dict:typing.Mapping[int, typing.Mapping[int, int]]) -> None:
     ...
 
+def str_len(x:str) -> int:
+    r"""
+    Returns the length of the string.
+    """
+    ...
+
 def sum(v:typing.Sequence[int]) -> int:
     r"""
     Returns the sum of two numbers as a string.

--- a/examples/pure/src/lib.rs
+++ b/examples/pure/src/lib.rs
@@ -61,6 +61,13 @@ fn create_a(x: usize) -> A {
 
 create_exception!(pure, MyError, PyRuntimeError);
 
+/// Returns the length of the string.
+#[gen_stub_pyfunction]
+#[pyfunction]
+fn str_len(x: &str) -> PyResult<usize> {
+    Ok(x.len())
+}
+
 /// Initializes the Python module
 #[pymodule]
 fn pure(m: &Bound<PyModule>) -> PyResult<()> {
@@ -70,6 +77,7 @@ fn pure(m: &Bound<PyModule>) -> PyResult<()> {
     m.add_function(wrap_pyfunction!(create_dict, m)?)?;
     m.add_function(wrap_pyfunction!(read_dict, m)?)?;
     m.add_function(wrap_pyfunction!(create_a, m)?)?;
+    m.add_function(wrap_pyfunction!(str_len, m)?)?;
     Ok(())
 }
 

